### ISSUE TITLE
Table rebuild instructions are wrong.

### ISSUE THREAD

**Issue by Yoshi2889** [dotnet/EntityFramework.Docs#2227]
The SQLite instructions for performing other operations are wrong according to the SQLite documentation (which is even directly linked in this article):
https://sqlite.org/lang_altertable.html#otheralter

Especially of note is the Caution note in this section.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: dbc4d900-28a8-9c04-2443-7407f15a684f
* Version Independent ID: 545bb0b5-f15b-584e-cef7-7283a0e0729f
* Content: [SQLite Database Provider - Limitations - EF Core](https://docs.microsoft.com/en-us/ef/core/providers/sqlite/limitations#feedback)
* Content Source: [entity-framework/core/providers/sqlite/limitations.md](https://github.com/dotnet/EntityFramework.Docs/blob/master/entity-framework/core/providers/sqlite/limitations.md)
* Product: **entity-framework**
* Technology: **entity-framework-core**
* GitHub Login: @rowanmiller
* Microsoft Alias: **avickers**

**Comment by ajcvickers** [comment]
@Yoshi2889 We used these instructions as the starting point for rebuilds. What, specifically, is it that EF is doing wrong?

**Comment by Yoshi2889** [comment]
The EF docs say the following:

> A table rebuild involves renaming the existing table, creating a new table, copying data to the new table, and dropping the old table.

However, the SQLite docs explicitly mention the following, with this graphic attached:
![image](https://user-images.githubusercontent.com/1280380/77958659-e2a76880-72c4-11ea-8864-a15c0d481685.png)
> Caution: Take care to follow the procedure above precisely. The boxes below summarize two procedures for modifying a table definition. At first glance, they both appear to accomplish the same thing. However, the procedure on the right does not always work, especially with the enhanced rename table capabilities added by versions 3.25.0 and 3.26.0. In the procedure on the right, the initial rename of the table to a temporary name might corrupt references to that table in triggers, views, and foreign key constraints. The safe procedure on the left constructs the revised table definition using a new temporary name, then renames the table into its final name, which does not break links.

**Comment by ajcvickers** [comment]
@bricelam to take a look.

**Comment by bricelam** [comment]
Agreed. Renaming the original table first is a bad idea.

I wonder if we should add a snippet like this to the docs:

```cs
// Create new table with desired schema
migrationBuilder.CreateTable(
    name: "new_Table1",
    ...);

// Copy data
migrationBuilder.Sql(@"
    INSERT INTO new_Table1 (Id, ...)
    -- TODO: Use IFNULL(Column1, '') to specify a default value for columns that
    --       are now nullable: false
    SELECT Id, ...
    FROM Table1;
");

// Suspend foreign keys
migrationBuilder.Sql("PRAGMA foreign_keys = 0;", suppressTransaction: true);

// Drop old table
migrationBuilder.DropTable(
    name: "Table1");

// Rename new into old
migrationBuilder.RenameTable(
    name: "new_Table1",
    newName: "Table1");

// Re-enable foreign keys
migrationBuilder.Sql("PRAGMA foreign_keys = 1;", suppressTransaction: true);

// Rebuild indexes
migrationBuilder.CreateIndex(
    name: "IX_Table1_Column1",
    table: "Table1",
    column: "Column1");
...
```

**Comment by bricelam** [comment]
Fixed by #2506
